`reqwest` is always used but marked as an optional dependency

**Describe the bug**
We always depend on the `reqwest` create regardless of feature flags, but it is marked as an optional dependency and only included if the `telemetry` feature is enabled. Thus, we get a build failure if we try to build without the telemetry feature.

**To Reproduce**
Steps to reproduce the behavior:
1. `cargo build --no-default-features --features streaming`
2. <details>
   <summary>Compile Error</summary>

   ```
      error[E0433]: failed to resolve: use of unresolved module or unlinked crate `reqwest`
       --> src/network.rs:1342:18
        |
   1342 |     let client = reqwest::Client::new();
        |                  ^^^^^^^ use of unresolved module or unlinked crate `reqwest`
        |
        = help: if you wanted to use a crate named `reqwest`, use `cargo add reqwest` to add it to your `Cargo.toml`

   error[E0433]: failed to resolve: use of unresolved module or unlinked crate `reqwest`
       --> src/network.rs:2877:18
        |
   2877 |     let client = reqwest::Client::new();
        |                  ^^^^^^^ use of unresolved module or unlinked crate `reqwest`
        |
        = help: if you wanted to use a crate named `reqwest`, use `cargo add reqwest` to add it to your `Cargo.toml`

   error[E0282]: type annotations needed for `Result<_, _>`
       --> src/network.rs:1351:9
        |
   1351 |     let result = client
        |         ^^^^^^
   ...
   1361 |         if response.status().is_success() {
        |            -------- type must be known at this point
        |
   help: consider giving `result` an explicit type, where the placeholders `_` are specified
        |
   1351 |     let result: Result<_, E> = client
        |               ++++++++++++++

   error[E0282]: type annotations needed
       --> src/network.rs:1375:28
        |
   1375 |             let error_text = response
        |  ____________________________^
   1376 | |             .text()
   1377 | |             .await
        | |__________________^ cannot infer type

   error[E0282]: type annotations needed
       --> src/network.rs:2891:31
        |
   2891 |         if let Ok(lrc_resp) = resp.json::<LrcResponse>().await {
        |                               ^^^^ cannot infer type

   error[E0277]: the size for values of type `str` cannot be known at compilation time
       --> src/network.rs:2892:18
        |
   2892 |           if let Some(synced) = lrc_resp.syncedLyrics {
        |                  ^^^^^^^^^^^^ doesn't have a size known at compile-time
        |
        = help: the trait `Sized` is not implemented for `str`
   note: required by a bound in `std::prelude::v1::Some`
       --> /build/rustc-1.92.0-src/library/core/src/option.rs:607:5

   Some errors have detailed explanations: E0277, E0282, E0433.
   For more information about an error, try `rustc --explain E0277`.
   error: could not compile `spotatui` (bin "spotatui") due to 6 previous errors
   ```
   </details>

**Expected behavior**
The code compiles.

**Desktop (please complete the following information):**
 - OS: NixOS
 - Terminal: foot
 - Version: master @ 864c6196e057552853439083705dc00ef120e3bd



**Additional context**
As a workaround, we can pass `reqwest` as a feature flag:
`cargo build --no-default-features streaming,reqwest`

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ anyhow = "1.0.43"
 chrono = "0.4.43"
 open = "5.3"
 self_update = { version = "0.42", features = ["archive-tar", "archive-zip", "compression-flate2", "compression-zip-deflate"] }
-reqwest = { version = "0.12", features = ["json", "rustls-tls"], default-features = false, optional = true }
+reqwest = { version = "0.12", features = ["json", "rustls-tls"], default-features = false }
 openssl = { version = "0.10", optional = true }
 cpal = { version = "0.17", optional = true }
 realfft = { version = "3.4", optional = true }
@@ -90,7 +90,7 @@ block2 = { version = "0.6", optional = true }
 
 [features]
 default = ["telemetry", "streaming", "audio-viz-cpal", "mpris", "macos-media", "discord-rpc"]
-telemetry = ["reqwest", "reqwest/rustls-tls"]
+telemetry = []
 streaming = ["librespot-core", "librespot-playback", "librespot-connect", "librespot-oauth", "librespot-metadata"]
 # Audio backend features
 alsa-backend = ["streaming", "librespot-playback/alsa-backend"]

